plugin/sampling/strategystore/static/factory_test.go

```diff
@@ -39,3 +45,43 @@ func TestFactory(t *testing.T) {
 	_, err := f.CreateStrategyStore()
 	assert.NoError(t, err)
 }
+
+func TestAutoReload(t *testing.T) {
+	// copy from fixtures/strategies.json
+	srcFile, dstFile := "fixtures/strategies.json", "fixtures/strategies_for_reload.json"
+	srcBytes, err := ioutil.ReadFile(srcFile)
+	require.NoError(t, err)
+	err = ioutil.WriteFile(dstFile, srcBytes, 0644)
+	require.NoError(t, err)
+
+	f := NewFactory()
+	v, command := config.Viperize(f.AddFlags)
+	_ = command.ParseFlags([]string{"--sampling.strategies-file=" + dstFile, "--sampling.strategies-reload-interval=50ms"})
+	f.InitFromViper(v)
+
+	// Test reading strategies from a file
+	//ctx, canf := context.WithCancel(context.Background())
+	//defer canf()
+
+	store, err := f.CreateStrategyStore()
```

Comment:
This leaks a goroutine. There needs to be a way to shut down cleanly.

Reply:
if we move reloading loop into storage, the test should move there as well (factory should only test parsing the options)